File: src/main/java/fr/ribesg/bukkit/pure/util/RemapAllMain.java
package fr.ribesg.bukkit.pure.util;

import fr.ribesg.bukkit.pure.*;

import java.io.IOException;
import java.nio.file.*;
import java.util.concurrent.*;
import java.util.logging.Level;

/**
 * @author Ribesg
 */
public final class RemapAllMain {

    /**
     * Download, remap and hash all known MC version.
     */
    public static void main(final String[] args) throws Throwable {
        final ExecutorService executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() - 1);
        for (final MCVersion v : MCVersion.values()) {
            executor.submit(new Runnable() {
                @Override
                public void run() {
                    try {
                        MCJarHandler.require(v, false);
                    } catch (final IOException e) {
                        Pure.logger().log(Level.SEVERE, "Failed to require " + v, e);
                    }
                }
            });
            Thread.sleep(250);
        }
        executor.shutdown();
        executor.awaitTermination(5, TimeUnit.MINUTES);
        try (final DirectoryStream<Path> s = Files.newDirectoryStream(Paths.get("jars"))) {
            for (final Path p : s) {
                Pure.logger().info(p.getFileName() + "\n\t" + HashUtils.hashSha256(p));
            }
        }
    }
}

File: src/main/kotlin/fr/ribesg/bukkit/pure/util/remap/RemapAllMain.kt
package fr.ribesg.bukkit.pure.util.remap

import fr.ribesg.bukkit.pure.MCJarHandler
import fr.ribesg.bukkit.pure.MCVersion
import fr.ribesg.bukkit.pure.Pure
import fr.ribesg.bukkit.pure.util.FileUtils
import fr.ribesg.bukkit.pure.util.HashUtils
import java.io.IOException
import java.net.InetSocketAddress
import java.net.Proxy
import java.net.Proxy.Type
import java.nio.file.Files
import java.nio.file.Paths
import java.util.concurrent.Executors
import java.util.concurrent.TimeUnit
import java.util.logging.Level

/**
 * @author Ribesg
 */

/**
 * Download, remap and hash all known MC version.
 */
public fun main(args: Array<String>) {
    try {
        if (args.size() == 2) {
            val proxyUrl = args[0]
            val proxyPort = try {
                Integer.parseInt(args[1])
            } catch (e: NumberFormatException) {
                throw IllegalArgumentException("Usage: [proxyUrl proxyPort]")
            }
            FileUtils.proxy = Proxy(Type.HTTP, InetSocketAddress(proxyUrl, proxyPort))
        } else if (args.size() != 0) {
            throw IllegalArgumentException("Usage: [proxyUrl proxyPort]")
        }
    } catch(e: IllegalArgumentException) {
        Pure.logger().log(Level.SEVERE, "Invalid arguments", e)
    }

    val executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() - 1)
    for (v in MCVersion.values()) {
        executor.submit({
            try {
                MCJarHandler.require(v, false)
            } catch (e: IOException) {
                Pure.logger().log(Level.SEVERE, "Failed to require " + v, e)
            }
        })
        Thread.sleep(250)
    }
    executor.shutdown()
    executor.awaitTermination(5, TimeUnit.MINUTES)
    Files.newDirectoryStream(Paths.get("jars")).use { directory ->
        for (file in directory) {
            Pure.logger().info(HashUtils.hashSha256(file) + " - " + file.getFileName())
        }
    }
}
